lib/ret_web/controllers/api/v1/app_config_controller.ex

```diff
@@ -1,18 +1,31 @@
 defmodule RetWeb.Api.V1.AppConfigController do
   use RetWeb, :controller
-  alias Ret.{Repo, AppConfig}
+  alias Ret.{Repo, AppConfig, Storage, OwnedFile}
 
   def create(conn, _params) do
     {:ok, body, conn} = conn |> Plug.Conn.read_body()
 
     # We expect the request body to be a json object where the leaf nodes are the config values.
     collapsed_config = body |> Poison.decode!() |> AppConfig.collapse()
 
+    account = Guardian.Plug.current_resource(conn)
+
     collapsed_config
     |> Enum.each(fn {key, val} ->
-      (AppConfig |> Repo.get_by(key: key) || %AppConfig{})
-      |> AppConfig.changeset(%{"key" => key, "value" => val})
-      |> Repo.insert_or_update!()
+      app_config = AppConfig |> Repo.get_by(key: key) || %AppConfig{}
+
+      app_config =
+        case val do
+          %{"file_id" => file_id, "meta" => %{"access_token" => access_token, "promotion_token" => promotion_token}} ->
+            {:ok, owned_file} = Storage.promote(file_id, access_token, promotion_token, account)
+            file_uri = owned_file |> OwnedFile.uri_for() |> URI.to_string()
```

Comment:
i think we should defer URL generation to the file at app config GET time, not here, since that URL structure may change. the 'value' of the config is the file itself semantically.